### PR TITLE
Docs revamp

### PR DESCRIPTION
**Pull request by abkfenris**
Start of a big revamp of the docs.
- Uses Pydata theme
- Enables MyST markdown and converts a lot of the docs to it (rst heading levels and git conflict formatting conflict, which was a fun discovery)
- Re-arranges things along the Diátaxis framework lines, or at least how they are being used in the PyData space
- Adds the vision or 'Why Xpublish' page
- Breaks the tutorial into sections and scaffolds sections for use of plugins

### CODE REVIEW

**Comment by abkfenris** [comment]
[RTD Preview](https://xpublish--180.org.readthedocs.build/en/180/index.html)

File: docs/source/getting-started/tutorial/introduction.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+- `/keys`: returns a list of variable keys, i.e., those returned by {attr}`xarray.Dataset.variables`.
+- `/info`: returns a JSON dictionary summary of a Dataset variables and attributes, similar to {meth}`xarray.Dataset.info`.
+- `/dict`: returns a JSON dictionary of the full dataset.
+- `/versions`: returns JSON dictionary of the versions of Python, Xarray and related libraries on the server side, similar to {func}`xarray.show_versions`.
+
+The application also provides data access through a [Zarr] compatible API with the
+following endpoints:
+
+- `/.zmetadata`: returns a JSON dictionary representing the consolidated Zarr metadata.
+- `/{var}/{key}`: returns a single chunk of an array.
```
Comment (jhamman) [inline docs/source/getting-started/tutorial/introduction.md:77]: update after #188 goes in

File: docs/source/getting-started/tutorial/introduction.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+that implements an HTTPStore. In Python, this can be done with `fsspec`:
+
+```python
+import zarr
+from fsspec.implementations.http import HTTPFileSystem
+
+fs = HTTPFileSystem()
+
+# The URL 'http://0.0.0.0:9000' here serves one dataset
+http_map = fs.get_mapper('http://0.0.0.0:9000')
```
Comment (jhamman) [inline docs/source/getting-started/tutorial/introduction.md:99]: update after #188 goes in

File: docs/source/getting-started/tutorial/serving-multiple-datasets.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Serving multiple datasets
+
+Xpublish also lets you serve multiple datasets via one FastAPI application. You
+just need to provide a mapping (dictionary) when creating a
+{class}`~xpublish.Rest` instance, e.g.,
+
+```python
+ds2 = xr.tutorial.open_dataset('rasm')
```
Comment (jhamman) [inline docs/source/getting-started/tutorial/serving-multiple-datasets.md:8]:
```suggestion
ds = xr.tutorial.open_dataset('air_temperature')
ds2 = xr.tutorial.open_dataset('rasm')
```

File: docs/source/getting-started/why-xpublish.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Why Xpublish
+
+Xarray has been key to new ways of interacting with and storing multi-dimensional datasets.
+It's grown a huge ecosystem and is constantly making forward progress.
```
Comment (jhamman) [inline docs/source/getting-started/why-xpublish.md:4]:
```suggestion
Xarray provides an intuitive API on top of a foundational data model, labeled arrays and datasets.
This API and data model has formed the basis for a large and growing ecosystem of tools.
```

File: docs/source/index.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+Don't mind the mess, our docs are currently being restructured.
+```
+
+**Useful links:** [Installation](getting-started/installation) | [Source Repository](https://github.com/xpublish-community/xpublish/) | [Issue Tracker](https://github.com/xpublish-community/xpublish/issues) | [Q&A Support](https://github.com/xpublish-community/xpublish/discussions/categories/q-a?discussions_q=category%3AQ%26A+) | [Slack Channel](./ecosystem/index.md#slack)
+
+Xpublish is:
+
+::::{grid} 3
+
+:::{grid-item-card} A quick way to serve a single Xarray dataset
```
Comment (jhamman) [inline docs/source/index.md:13]:
```suggestion
:::{grid-item-card} A quick way to serve a single Xarray dataset over HTTP using FastAPI
```

File: docs/source/index.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+:link: getting-started/tutorial/introduction
+
+Add a REST API to your dataset with `ds.rest.serve()`!
```
Comment (jhamman) [inline docs/source/index.md:16]: this feels duplicative

File: docs/source/index.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+Those datasets can be accessed from various kinds of client applications, e.g.,
+from within Python using Zarr and fsspec.
+
+```python
+import xarray as xr
+import zarr
+from fsspec.implementations.http import HTTPFileSystem
+
+fs = HTTPFileSystem()
+http_map = fs.get_mapper('http://0.0.0.0:9000')
```
Comment (jhamman) [inline docs/source/index.md:98]: update following #188

File: docs/source/index.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+application.
+
+The data and/or metadata in the Xarray Datasets can be exposed in various forms
+through pluggable REST API endpoints. Efficient, on-demand delivery of large
+datasets may be enabled with Dask on the server-side.
+
+We are exploring applications of Xpublish that include:
+
+- publish on-demand or derived data products
+- turning xarray objects into streaming services (e.g. OPeNDAP)
```
Comment (jhamman) [inline docs/source/index.md:119]: this could probably be updated (or removed)

**Review by jhamman — COMMENTED** [review]
This is a great start! I didn't make it all the way through but kudos for taking the time to make this happen. 🙌

File: docs/source/index.md (hunk 6 above)
Comment (abkfenris) [reply docs/source/index.md:16]: The whole card, or 'Add a REST API to your dataset with `ds.rest.serve()`?

How about something like 'Get started with `ds.rest.serve()` to explore serving data with Xpublish'?

File: docs/source/index.md (hunk 8 above)
Comment (abkfenris) [reply docs/source/index.md:120]: Yup. I've been slowly rebuilding the index from the top down, but keeping old stuff around till I'm sure I've replaced it.

Comment (abkfenris) [reply docs/source/index.md]: I went and removed that whole section.

**Pull request merged**